lib/elixir/lib/calendar/date.ex

```diff
@@ -56,6 +56,27 @@ defmodule Date do
   @type t :: %Date{year: Calendar.year, month: Calendar.month,
                    day: Calendar.day, calendar: Calendar.calendar}
 
+
+  @doc """
+  Returns a Date.Range
+
+  ## Examples
+
+      iex> Date.range(~D[2000-01-01], ~D[2001-01-01])
+      %Date.Range{
+        first: ~D[2000-01-01],
+        last: ~D[2001-01-01],
+        first_rata_die: {730120, {0, 86400000000}},
+        last_rata_die: {730486, {0, 86400000000}},
+        kind: :asc
+      }
+
+  """
+  @spec range(Date.t, Date.t) :: Date.Range.t
+  def range(first, last) do
+    Date.Range.new(first, last)
```

Comment:
Please build the range directly here instead of calling `new`. I.e. move the contents of `Date.Range.new` to here. This way you won't need to make to_rata_die non-private as well.